impl/maven-impl/src/test/java/org/apache/maven/impl/DefaultModelXmlFactoryTest.java

```diff
@@ -122,4 +126,91 @@ void testMalformedModelVersion() throws Exception {
         Model model = factory.read(request);
         assertEquals("invalid.version", model.getModelVersion());
     }
+
+    @Test
+    void testWriteModelWithoutInputLocationTracking() {
+        String model =
+                """
+                <project xmlns="http://maven.apache.org/POM/4.0.0">
```

Comment:
I don't think this works in Java 11. Are we requiring Java 17 now?

Reply:
Yes, Maven 4 has switched to 17 a while ago, see https://github.com/apache/maven/issues/10806